app/src/main/java/com/example/memeitupapp/ui/mainmemes/MainMemesActivity.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.memeitupapp.ui.mainmemes
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.viewpager2.widget.ViewPager2
+import com.example.memeitupapp.R
+import com.example.memeitupapp.ui.adapter.ViewPagerFragmentAdapter
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayoutMediator
+import kotlinx.android.synthetic.main.activity_main.pager
+import kotlinx.android.synthetic.main.activity_main.tabLayout
+
+class MainMemesActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        initViewPagerWithFragments()
+    }
+
+    private fun initViewPagerWithFragments() {
+        val viewPager: ViewPager2 = pager
+        val adapter = ViewPagerFragmentAdapter(supportFragmentManager, lifecycle)
+        viewPager.adapter = adapter
+
+        val tabLayout: TabLayout = tabLayout
+        val names: MutableList<String> = mutableListOf(GRID_OF_MEMES, LIST_OF_MEMES)
+        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
+            tab.text = names[position]
+        }.attach()
+    }
+
+    companion object {
+        private const val GRID_OF_MEMES = "Grid of Memes"
```

Comment:
do not put this here, use string resources

Reply:
now i'm using string resources, thanks